### DISCUSSION TITLE
Important Links

### DISCUSSION THREAD

**Discussion by TimZiemer** [ifsm/pymus#2] [Announcements]
Everyone can share links he or she finds valuable. 

Our Zoom Link is: 
https://uni-hamburg.zoom.us/j/64686101884?pwd=anl5RS9sTE0wazJ4MmhEalRLK0hIQT09

The passcode is:
95593723

Maybe you found helpful resources for learning python programming or musicology related topics. Feel free to share them.

**Comment by Teagum** [comment]
## Python documentation

The Python community maintains an exceptionally well written [documentation](https://docs.python.org/3/). You'll find there everything necessary for your day-to-day Python experience. So, no need to rely on third-party, ad contaminated, badly written, so called tutorial.

- [Python Tutorial](https://docs.python.org/3/tutorial/index.html) - start here if you're new to Python 
- [Python Standard Library](https://docs.python.org/3/library/index.html) - go there if you want know about what's included in Python and how you can use it 
- [Python Language Reference](https://docs.python.org/3/reference/index.html) - go there if you want to learn about Python's structure, syntax and semantic

## The scientific stack

The Python community also provides a large scientific environment, including libraries for most applications and use case. Typically, you'll spend most of the time with the following packages:

- [NumPy](https://numpy.org/) - the fundamental package for scientific computing and array manipulation
- [SciPy](https://docs.scipy.org/doc/scipy/index.html) - algorithms for signal processing, statistics, linear algebra, and many more
- [pandas](https://pandas.pydata.org/) - a high-level analysis tool for tabular data
- [matplotlib](https://matplotlib.org/) - the fundamental package for 2d graphics and publication-ready plots


## Notebooks

Notebooks are a very convenient way for interactively working with your data. [Project Jupyter](https://jupyter.org/) provides a popular, language-agnostic notebook interface that runs in your favorite browser. In fact, Jupyter with Python is, the de-facto standard tool for the whole project lifecycle in Data Science and AI. You will typically use JupyterLab for prototyping, data cleaning, model training, evaluation and presentation. You can even prepare portable slides for your next conference gig, that run on every operating system. Go check it out!

Note that there are currently two interfaces: JupyterLab and Jupyter Notebook. You should use JupyterLab. Jupyter Notebook is the legacy or classic interface that soon will be fully replaced by JupyterLab.


## Don't reinvent the wheel

Most novice programmers will typically spend many hours doing things that

* somebody else did before
* somebody did way better

As a general rule of thumb, before you write _anything_, take a look at the standard library to see if there is already a solution that suits your needs. If you don't find your desired algorithm, check out the scientific stack; they'll have it in 95% of all cases. If you can't find it there, look again. It really is there, you just didn't frame your problem correctly, because you probably thought to narrowly and specific - go abstract. If that should fail, too, then and only then you'll role out your own algorithm.

Have fun!

**Comment by TimZiemer** [comment]
Written Git and Github tutorials can be found on
https://github.com/ifsm/pymus/assets/94977618/933fbccd-fbcd-454e-a1f0-ab600ff1e332
https://github.com/ifsm/pymus/assets/94977618/51b68917-27cf-47b2-8228-938355ec64f3
https://github.com/ifsm/pymus/assets/94977618/fd99a403-e0e3-4cfd-9683-e81efdf0364c

Video tutorials are available at
https://www.youtube.com/playlist?list=PLRqwX-V7Uu6ZF9C0YMKuns9sLDzK6zoiV
https://www.youtube.com/playlist?list=PL4cUxeGkcC9goXbgTDQ0n_4TBzOO0ocPR
https://www.udemy.com/course/git-and-github-crash-course-creating-a-repository-from-scratch
https://www.youtube.com/playlist?list=PLhW3qG5bs-L8OlICbNX9u4MZ3rAt5c5GG
https://www.youtube.com/playlist?list=PLWKjhJtqVAbkFiqHnNaxpOPhh9tSWMXIF
